cmd/influx_inspect/verify.go

```diff
@@ -0,0 +1,64 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"text/tabwriter"
+
+	"github.com/influxdb/influxdb/tsdb/engine/tsm1"
+)
+
+func cmdVerify(path string) {
+	dataPath := filepath.Join(path, "data")
+
+	brokenBlocks := 0
+	totalBlocks := 0
+
+	// No need to do this in a loop
+	ext := fmt.Sprintf(".%s", tsm1.TSMFileExtension)
+
+	// Get all TSM files by walking through the data dir
+	files := []string{}
+	err := filepath.Walk(dataPath, func(path string, f os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if filepath.Ext(path) == ext {
+			files = append(files, path)
+		}
+		return nil
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	tw := tabwriter.NewWriter(os.Stdout, 16, 8, 0, '\t', 0)
+
+	// Verify the checksums of every block in every file
+	for _, f := range files {
+		file, err := os.OpenFile(f, os.O_RDONLY, 0600)
+		if err != nil {
+			panic(err)
+		}
+
+		reader, err := tsm1.NewTSMReader(file)
+		if err != nil {
+			panic(err)
+		}
+
+		for key, checksums := range reader.ValidateChecksum() {
+			for block, valid := range checksums {
+				totalBlocks++
+				if !valid {
+					brokenBlocks++
+					fmt.Fprintf(tw, "[Broken Block] in File: %s\t\n Key: %s\t Block: %s\t\n", f, key, block)
+					break
```

Comment:
Is there maybe value in knowing how many blocks are broken, i.e., not breaking when you find the first block? Or is it the case that once one block is broken, rest would be broken anyway?